### PR TITLE
Add preliminary support for linearity correction

### PR DESCRIPTION
**Pull request by r-owen**
Update DecamMapper.paf to add dataset type "linearizer"
Add bin.src/ingestLinearizeDecam.py to ingest lookup table
linearizers from the official DECam linearization FITS file.
Update camera generator to set linearity type and coefficients
in amp info tables.
Use the new camera generating code to generate new amp info tables.
This is preliminary because the `linearize` dataset type is not
persisted as a calibration product, so it has no table containing
date range and such. Also ingest is handled by a separate script
instead of as part of calibration ingest.

### CODE REVIEW

File: bin.src/ingestLinearizeDecam.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+def ingestLinearizeDecam(inPath, repoPath):
+    """Convert the specified DECam linearity FITS table to standard LSST format
+
+    Details:
+    - Input format is one table per CCD, HDU is amplifier number,
+        the table has 3 columns: ADU, ADU_LINEAR_A, ADU_LINEAR_B.
+        The values of ADU contiguous (0, 1, 2...) but check and error out if not.
+        The values of the latter two are replacements (0+delta0, 1+delta1, 2+delta2...)
+        and this is converted to offsets for the LSST linearization tables (delta0, delta1, delta2...)
+    - Output is to %(path)s/linearity/_%02(ccdnum)d.pickle
```
Comment (hchiang2) [inline bin.src/ingestLinearizeDecam.py]: The template has changed

File: bin.src/ingestLinearizeDecam.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - The row indices for the linearization lookup table are (row index=amp name): 0=A, 1=B
+
+    @param[in] inPath  path to input table (one HDU per amplifier)
+    @param[in] butler  butler to output repo
```
Comment (hchiang2) [inline bin.src/ingestLinearizeDecam.py]: butler is not a input parameter here

File: bin.src/ingestLinearizeDecam.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        uncorr = fromData["ADU"]
+        if not np.allclose(uncorr, np.arange(len(fromData))):
+            raise RuntimeError("ADU data not a range of integers starting at 0")
+        for i, ampName in enumerate("AB"):
+            # convert DECam replacement table to LSST offset table
+            print("DECam table for %s=%s..." % (ampName, fromData["ADU_LINEAR_" + ampName][0:5],))
+            lsstTable[i,:] = fromData["ADU_LINEAR_" + ampName] - uncorr
+            print("LSST  table for %s=%s..." % (ampName, lsstTable[i,0:5],))
+        linearizer = LinearizeLookupTable(table=lsstTable, detector=detector)
+        dataId = dict(path=repoPath, ccdnum=ccdnum)
```
Comment (hchiang2) [inline bin.src/ingestLinearizeDecam.py]: "path" is not needed with the current template

File: bin.src/ingestLinearizeDecam.py (hunk 1 above)
Comment (r-owen) [inline bin.src/ingestLinearizeDecam.py]: Oops. I don't know why I ever had that leading underscore!

File: bin.src/ingestLinearizeDecam.py (hunk 2 above)
Comment (r-owen) [inline bin.src/ingestLinearizeDecam.py]: Good catch

File: bin.src/ingestLinearizeDecam.py (hunk 1 above)
Comment (r-owen) [inline bin.src/ingestLinearizeDecam.py]: In any case it's a bad idea to duplicate information from `DecamMapper.paf` so I changed this to:

    Output is a set of LinearizeLookupTable instances, one per CCD saved as dataset type "linearizer"

File: bin.src/ingestLinearizeDecam.py (hunk 3 above)
Comment (r-owen) [inline bin.src/ingestLinearizeDecam.py]: Good point. I had missed that change when you fixed the linearizer entry in the .paf file

File: decam/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+obs_decam/decam
+===============
+
+This directory includes camera geometry and amplifier characteristics data
+that is used by the butler to create the "camera" and "linearizer" data products.
+The linearity table `linearity_table_v*.fits` is a standard DECam calibration file from
+[DECam Community Pipeline Calibration Files](http://www.ctio.noao.edu/noao/content/decam-calibration-files).
+Building the `obs_decam` package converts the linearity table to LSST linearizers `linearize/*.fits`
+by running `makeLinearize.py`.
```
Comment (hchiang2) [inline decam/README.md]: linearize/*.fits --> linearizer/*.fits
makeLinearize.py --> makeLinearizer.py

Comment (r-owen) [inline decam/README.md]: Good catch

**Pull request merged**